algorithms/kernel/kernel_function/oneapi/kernel_function_rbf_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,184 @@
+                                                                                                 NumericTable * r, const ParameterBase * par)
+{
+    //prepareData
+    services::Status status;
+
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & deviceInfo = context.getInfoDevice();
+
+    const size_t nVectors1 = a1->getNumberOfRows();
+    const size_t nVectors2 = a2->getNumberOfRows();
+
+    const size_t nFeatures1 = a1->getNumberOfColumns();
+    const size_t nFeatures2 = a2->getNumberOfColumns();
+
+    const Parameter * rbfPar    = static_cast<const Parameter *>(par);
+    const algorithmFPType coeff = algorithmFPType(-0.5 / (rbfPar->sigma * rbfPar->sigma));
+
+    BlockDescriptor<algorithmFPType> a1BD;
+    BlockDescriptor<algorithmFPType> a2BD;
+    BlockDescriptor<algorithmFPType> rBD;
+
+    const size_t startRows = 0;
+    DAAL_CHECK_STATUS(status, a1->getBlockOfRows(startRows, nVectors1, ReadWriteMode::readOnly, a1BD));
+    DAAL_CHECK_STATUS(status, a2->getBlockOfRows(startRows, nVectors2, ReadWriteMode::readOnly, a2BD));
+
+    DAAL_CHECK_STATUS(status, r->getBlockOfRows(startRows, nVectors1, ReadWriteMode::writeOnly, rBD));
+
+    const services::Buffer<algorithmFPType> a1Buf = a1BD.getBuffer();
+    const services::Buffer<algorithmFPType> a2Buf = a2BD.getBuffer();
+
+    services::Buffer<algorithmFPType> rBuf = rBD.getBuffer();
+
+    UniversalBuffer sqrA1U = context.allocate(TypeIds::id<algorithmFPType>(), nVectors1, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    UniversalBuffer sqrA2U = context.allocate(TypeIds::id<algorithmFPType>(), nVectors2, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    //compute
+    {
+        DAAL_ITTNOTIFY_SCOPED_TASK(KernelRBF.sumOfSquared);
+
+        Reducer::reduce(Reducer::BinaryOp::SUM_OF_SQUARES, Layout::RowMajor, a1Buf, sqrA1U, nVectors1, nFeatures1, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+        Reducer::reduce(Reducer::BinaryOp::SUM_OF_SQUARES, Layout::RowMajor, a2Buf, sqrA2U, nVectors2, nFeatures2, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    {
+        DAAL_ITTNOTIFY_SCOPED_TASK(KernelRBF.gemm);
+        DAAL_CHECK_STATUS(status, BlasGpu<algorithmFPType>::xgemm(math::Layout::RowMajor, math::Transpose::NoTrans, math::Transpose::Trans, nVectors1,
+                                                                  nVectors2, nFeatures1, algorithmFPType(-2.0), a1Buf, nFeatures1, 0, a2Buf,
+                                                                  nFeatures2, 0, algorithmFPType(0.0), rBuf, nVectors2, 0));
+    }
+
+    const services::Buffer<algorithmFPType> sqrA1Buff = sqrA1U.get<algorithmFPType>();
+    const services::Buffer<algorithmFPType> sqrA2Buff = sqrA2U.get<algorithmFPType>();
+
+    DAAL_CHECK_STATUS(status, computeRBF(sqrA1Buff, sqrA2Buff, nVectors2, coeff, rBuf, nVectors1, nVectors2));
```

Comment:
I see here potential problem with implicit type conversion. The 3-rd argument defined as `const uint32_t ld`, but passed here as `size_t` that leads to overflow. The overflow must be checked before `computeRBF` invocation and appropriate status code returned.